Automatic Update Feature

dalogue1 opened the pull request:
The following are the changes to add a feature to allow an installation to perform automatic updates without user intervention.

The basic changes are as follows:

- addition of a checkbox attribute in the Advanced Settings webpage to indicate the desire fr automatic update
- addition of the attribute in the configuration file generated on installation
- creation of a file to append to the crontab file for the automatic update schedule
- creation of a script to check for the status of the new AUTOMATIC_UPDATE attribute and update and restart if required
- modification of the installation of services to include the new scheduled task
- automatic_update.sh needs to be made executable

alexbelgium commented:
A facultative automatic update looks like a great feature indeed !

alexbelgium commented:
You should also update the "update_birdnet_snippets.sh" file to patch existing installations. This is how previous new cron jobs were added

dalogue1 commented:
I was wondering how that was done.  I will do that.

I came across the issue with the access permission of install_helpers.sh while working on this feature. https://github.com/Nachtzuster/BirdNET-Pi/issues/428

alexbelgium commented:
7 min after midnight seems a good idea ! It will be interesting when Nachtzuster looks at this also as he is much more knowledgeable than me on the in depth of birdnet-pi and code robustness

alexbelgium commented:
And simplifies management of issues as people have more probability to be on the latest code

Nachtzuster commented:
I would be fine with checking every 7-10days, that way it could stay just a cron job and stay simple

Nachtzuster commented on scripts/advanced.php:
```diff
@@ -459,6 +468,9 @@ function collectrtspUrls() {
       <label for="silence_update_indicator">Silence Update Indicator: </label>
       <input type="checkbox" name="silence_update_indicator" <?php if($newconfig['SILENCE_UPDATE_INDICATOR'] == 1) { echo "checked"; };?> ><br>
       <p>This allows you to quiet the display of how many commits your installation is behind by relative to the Github repo. This number appears next to "Tools" when you're 50 or more commits behind.</p>
+      <label for="automatic_update">Automatic Update After Midnight: </label>
```
the spacing in this Options section is kinda odd, if you feel up to it, you may fix it
<img width="1403" height="334" alt="image" src="https://github.com/user-attachments/assets/9f829fdf-0840-441d-a508-88edd822fa65" />

Nachtzuster reviewed (changes requested):
You need to add something like: https://github.com/Nachtzuster/BirdNET-Pi/blob/375c082ffb737ec01d99913df1b122dda44c6f61/scripts/update_birdnet_snippets.sh#L107C1-L109C3 , so existing installs get the setting too.

dalogue1 commented:
I made the following changes:

1. added the code to create the AUTOMATIC_UPDATE option to existing installations.
2. removed the unnecessary call to restart_services.sh
3. cleaned up the line spacing in the Options section of the settings table
